src/main/java/core/basesyntax/operations/SupplyOperation.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> storage, FruitTransaction transaction) {
+        storage.put(transaction.getFruit(),
+                storage.getOrDefault(transaction.getFruit(), 0)
+                        + transaction.getQuantity());
+    }

```

Comment:
It's good practice to validate the input parameters to ensure they are not null to prevent potential `NullPointerException`. Consider adding a check for both `storage` and `transaction` at the beginning of the `apply` method.